### PR TITLE
docs: chains & memory fixes

### PR DESCRIPTION
**Pull request by seamusp**
Various improvements to the Chains & Memory sections of the documentation including formatting, spelling, and grammar fixes to improve readability.

### CODE REVIEW

File: docs/docs_skeleton/docs/modules/chains/document/refine.mdx [hunk 1]
```diff
@@ -5,8 +5,8 @@ sidebar_position: 1
 
 The refine documents chain constructs a response by looping over the input documents and iteratively updating its answer. For each document, it passes all non-document inputs, the current document, and the latest intermediate answer to an LLM chain to get a new answer.
 
-Since the Refine chain only passes a single document to the LLM at a time, it is well-suited for tasks that require analyzing more documents than can fit in the model's context.
-The obvious tradeoff is that this chain will make far more LLM calls than, for example, the Stuff documents chain.
-There are also certain tasks which are difficult to accomplish iteratively. For example, the Refine chain can perform poorly when documents frequently cross-reference one another or when a task requires detailed information from many documents.
+Since the refine chain only passes a single document to the LLM at a time, it is well-suited for tasks that require analyzing more documents than can fit in the model's context.
```
Comment (leo-gan) [inline docs/docs_skeleton/docs/modules/chains/document/refine.mdx:8]: Refine, Stuff - those are terms, names of the chains. They should be capitalized.
Also L9, L10

File: docs/extras/modules/chains/how_to/openai_functions.ipynb [hunk 2]
```diff
@@ -342,7 +342,7 @@
     "### Using Python functions\n",
     "We can pass in functions as Pydantic classes, directly as OpenAI function dicts, or Python functions. To pass Python function in directly, we'll want to make sure our parameters have type hints, we have a docstring, and we use [Google Python style docstrings](https://google.github.io/styleguide/pyguide.html#doc-function-args) to describe the parameters.\n",
     "\n",
-    "**NOTE**: To use Python functions, make sure the function arguments are of primitive types (str, float, int, bool) or that they are Pydantic objects."
+    "**Note**: To use Python functions, make sure the function arguments are of primitive types (str, float, int, bool) or that they are Pydantic objects."
```
Comment (leo-gan) [inline docs/extras/modules/chains/how_to/openai_functions.ipynb:345]: We don't have rules for `note` formatting. All capital is OK, it catches attention well. Just saying.

File: docs/extras/modules/memory/adding_memory.ipynb [hunk 3]
```diff
@@ -7,9 +7,9 @@
     "tags": []
    },
    "source": [
-    "# How to add Memory to an LLMChain\n",
+    "# How to add memory to an LLMChain\n",
     "\n",
-    "This notebook goes over how to use the Memory class with an LLMChain. For the purposes of this walkthrough, we will add  the [ConversationBufferMemory](https://api.python.langchain.com/en/latest/memory/langchain.memory.buffer.ConversationBufferMemory.html#langchain.memory.buffer.ConversationBufferMemory) class, although this can be any memory class."
+    "This notebook goes over how to use a memory class with an `LLMChain`. For the purposes of this walkthrough, we will add  the [ConversationBufferMemory](https://api.python.langchain.com/en/latest/memory/langchain.memory.buffer.ConversationBufferMemory.html#langchain.memory.buffer.ConversationBufferMemory) class, although this can be any memory class."
```
Comment (leo-gan) [inline docs/extras/modules/memory/adding_memory.ipynb:12]: Memory is the proper name of the class.

File: docs/extras/modules/memory/adding_memory.ipynb [hunk 4]
```diff
@@ -32,7 +32,7 @@
    "id": "4b066ced",
    "metadata": {},
    "source": [
-    "The most important step is setting up the prompt correctly. In the below prompt, we have two input keys: one for the actual input, another for the input from the Memory class. Importantly, we make sure the keys in the PromptTemplate and the ConversationBufferMemory match up (`chat_history`)."
+    "The most important step is setting up the prompt correctly. In the below prompt, we have two input keys: one for the actual input, another for the input from the memory class. Importantly, we make sure the keys in the `PromptTemplate` and the `ConversationBufferMemory` match up (`chat_history`)."
```
Comment (leo-gan) [inline docs/extras/modules/memory/adding_memory.ipynb:35]: The same.

File: docs/extras/modules/memory/agent_with_memory.ipynb [hunk 5]
```diff
@@ -5,19 +5,19 @@
    "id": "fa6802ac",
    "metadata": {},
    "source": [
-    "# How to add Memory to an Agent\n",
+    "# How to add memory to an agent\n",
```
Comment (leo-gan) [inline docs/extras/modules/memory/agent_with_memory.ipynb:8]: Here it is about the specific classes with names: Memory and Agent, not generic memory and agent.

File: docs/extras/modules/memory/agent_with_memory.ipynb [hunk 6]
```diff
@@ -86,7 +86,7 @@
    "id": "0021675b",
    "metadata": {},
    "source": [
-    "We can now construct the LLMChain, with the Memory object, and then create the agent."
+    "We can now construct the `LLMChain`, with the memory object, and then create the agent."
```
Comment (leo-gan) [inline docs/extras/modules/memory/agent_with_memory.ipynb:89]: The same. I stop here to make the same comments. Many changes below about changing proper names of the classes to the generic words. Please, rollback them.

**Review by leo-gan — APPROVED** [review]
several nits with capitalization.
Otherwise, LGTM
Thanks @seamusp !

**Review by hwchase17 — APPROVED** [review]
this is fantastic!!! thanks so much

**Pull request merged**